### PR TITLE
Update meld to 3.16.0-r1,osx-9

### LINKED ISSUE #29398: Bug report: meld not in path after update
#### Description of issue

I installed meld a good while back, I'm pretty sure with `brew cask install meld`, and it's always worked fine from the command line. Today I discovered that `meld a b` no longer works because there is no longer a meld binary in my path. I think this is due to a meld update. The app is still there, and works when called directly. 

```zsh
➜  ~ which meld                                                                                             ~
meld not found

➜  ~ ls /usr/local/bin | grep meld                                                                  ~
# no results

➜  ~ ls /usr/local/Caskroom
docker     exa        meld       osxfuse    vagrant    vimr       virtualbox
➜  ~ ls /usr/local/Caskroom/meld
3.16.0-r1,osx-9
➜  ~ ls -a /usr/local/Caskroom/meld/3.16.0-r1,osx-9
.  ..

➜  ~ /Applications/Meld.app/Contents/MacOS/Meld a b                                                 ~
# Works
```

I tried to follow the issue guide, let me know if I did something wrong!

<details><summary>Output of `brew cask <command> --verbose`</summary>

```
➜  Homebrew git:(stable) brew cask reinstall --verbose meld                       /usr/local/Homebrew
==> Downloading https://github.com/yousseb/meld/releases/download/osx-9/meldmerge.dmg
/usr/bin/curl --remote-time --location --user-agent Homebrew/1.1.8-34-gfd48ec1c9 (Macintosh; Intel macOS 10.12.2) curl/7.51.0 --fail https://github.com/yousseb/meld/releases/download/osx-9/meldmerge.dmg -C 0 -o /Users/gib/Library/Caches/Homebrew/Cask/meld--3.16.0-r1,osx-9.dmg.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   590    0   590    0     0   1280      0 --:--:-- --:--:-- --:--:--  1282
100 64.8M  100 64.8M    0     0  6057k      0  0:00:10  0:00:10 --:--:-- 7003k
==> Verifying checksum for Cask meld
==> Removing App: '/Applications/Meld.app'
==> Moving App 'Meld.app' to '/Applications/Meld.app'
🍺  meld was successfully installed!
```
</details>

<details><summary>Output of `brew doctor`</summary>

```
➜  ~ brew doctor                                                                                    ~
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: The filesystem on / appears to be case-sensitive.
The default macOS filesystem is case-insensitive. Please report any apparent problems.

Warning: Putting non-prefixed findutils in your path can cause python builds to fail.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libmlrxapi1.dylib
  /usr/local/lib/libmwebsvc1.dylib
```
</details>

<details><summary>Output of `brew cask doctor`</summary>

```
➜  Homebrew git:(stable) brew cask doctor                                         /usr/local/Homebrew
==> macOS Release:
10.12.2
==> Hardware Architecture:
intel-64
==> Ruby Version:
2.0.0-p648
==> Ruby Path:
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
==> Homebrew Version:
Homebrew 1.1.8-34-gfd48ec1c9
Homebrew/homebrew-core (git revision 5daf; last commit 2017-01-25)
==> Homebrew Executable Path:
/usr/local/bin/brew
==> Homebrew Cellar Path:
/usr/local/Cellar
==> Homebrew Repository Path:
/usr/local/Homebrew
==> Homebrew Origin:
https://github.com/Homebrew/brew
==> Homebrew-Cask Version:
Homebrew-Cask 1.1.8-34-gfd48ec1c9
caskroom/homebrew-cask (git revision 723e; last commit 2017-01-25)
==> Homebrew-Cask Install Location:
<NONE>
==> Homebrew-Cask Staging Location:
/usr/local/Caskroom
==> Homebrew-Cask Cached Downloads:
/Users/gib/Library/Caches/Homebrew/Cask
0 files, 0B
==> Homebrew-Cask Default Tap Path:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
==> Homebrew-Cask Alternate Cask Taps:
==> Homebrew-Cask Default Tap Cask Count:
3539
==> Contents of $LOAD_PATH:
/usr/local/Homebrew/Library/Homebrew/cask/lib
/usr/local/Homebrew/Library/Homebrew
/Library/Ruby/Site/2.0.0
/Library/Ruby/Site/2.0.0/x86_64-darwin16
/Library/Ruby/Site/2.0.0/universal-darwin16
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/universal-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/universal-darwin16
==> Contents of $RUBYLIB Environment Variable:
<NONE>
==> Contents of $RUBYOPT Environment Variable:
<NONE>
==> Contents of $RUBYPATH Environment Variable:
<NONE>
==> Contents of $RBENV_VERSION Environment Variable:
<NONE>
==> Contents of $CHRUBY_VERSION Environment Variable:
<NONE>
==> Contents of $GEM_HOME Environment Variable:
<NONE>
==> Contents of $GEM_PATH Environment Variable:
<NONE>
==> Contents of $BUNDLE_PATH Environment Variable:
<NONE>
==> Contents of $PATH Environment Variable:
PATH="/usr/local/opt/ccache/libexec:/Users/gib/.cargo/bin:/Users/gib/.nvm/versions/node/v6.9.2/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/Homebrew/Library/Homebrew/cask/cmd:/usr/local/Homebrew/Library/Homebrew/shims/scm"
==> Contents of $SHELL Environment Variable:
SHELL="/usr/local/bin/zsh"
==> Contents of Locale Environment Variables:
LANG="en_GB.UTF-8"
LC_CTYPE="en_GB.UTF-8"
```
</details>

### PR DESCRIPTION
**Pull request by miccal**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

Closes https://github.com/caskroom/homebrew-cask/issues/29398

### CODE REVIEW

File: Casks/meld.rb [hunk 1]
```diff
@@ -5,9 +5,10 @@
   # github.com/yousseb/meld was verified as official when first introduced to the cask
   url "https://github.com/yousseb/meld/releases/download/#{version.after_comma}/meldmerge.dmg"
   appcast 'https://github.com/yousseb/meld/releases.atom',
-          checkpoint: '4c9b17456ea505a7490456e40b96b980ae3295ad844fed7e71ac1c091787c250'
+          checkpoint: '4f8b54087e9218c1195f74c5a70bbcfc32803fbbd92a3a90257437d7300a4f27'
   name 'Meld for OSX'
   homepage 'https://yousseb.github.io/meld/'
 
   app 'Meld.app'
+  binary "#{appdir}/Contents/MacOS/Meld", target: 'meld'
```
Comment (gibfahn) [inline Casks/meld.rb]: I think it needs to be `binary "#{appdir}/Meld.app/Contents/MacOS/Meld", target: 'meld'`

Comment (miccal) [inline Casks/meld.rb]: Thank you.

**Comment by PatrickBauer** [comment]
Tried the new changes, meld works now from the CLI. Thank you!

**Review by PatrickBauer — APPROVED** [review]

**Review by gibfahn — APPROVED** [review]
Works for me!

**Pull request merged**